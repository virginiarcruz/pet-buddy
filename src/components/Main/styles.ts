import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Container = styled.main`
  ${({ theme: { color } }) => css`
    background-color: ${color.mainBg};
    width: 100%;
    height: 100%;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 72px 16px 42px 16px;
    ${media.greaterThan('medium')`
      padding: 96px 156px 90px 156px;
    `}
  `}
`
export const Content = styled.section`
  ${({ theme: { color, border } }) => css`
    background-color: ${color.white};
    width: 100%;
    min-height: 200px;
    border-radius: ${border.radius.medium};
    margin-top: 56px;
    ${media.greaterThan('medium')`
      margin-top: 96px;
    `}
  `}
`
