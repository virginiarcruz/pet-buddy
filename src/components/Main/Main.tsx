import { Container, Content } from './styles'
import Logo from 'components/Logo'

const Main = () => (
  <Container data-testid="main-container">
    <Logo />
    <Content />
  </Container>
)

export default Main
