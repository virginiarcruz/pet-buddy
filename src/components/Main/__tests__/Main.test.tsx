import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Main from '../Main'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Main />)

    expect(
      screen.getByRole('img', { name: /Shelter Buddy logo/i })
    ).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    renderWithTheme(<Main />)
    expect(screen.getByTestId('main-container')).toHaveStyle({
      'background-color': '#F6F7FA'
    })
  })
})
