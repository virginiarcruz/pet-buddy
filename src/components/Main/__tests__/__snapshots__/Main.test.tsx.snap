// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #F6F7FA;
  width: 100%;
  height: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 72px 16px 42px 16px;
}

.c2 {
  background-color: #FFFFFF;
  width: 100%;
  min-height: 200px;
  border-radius: 12px;
  margin-top: 56px;
}

.c1 {
  width: 206px;
  height: 32px;
}

@media (min-width:768px) {
  .c0 {
    padding: 96px 156px 90px 156px;
  }
}

@media (min-width:768px) {
  .c2 {
    margin-top: 96px;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 283px;
    height: 44px;
  }
}

<div>
  <main
    class="c0"
    data-testid="main-container"
  >
    <img
      alt="Shelter Buddy logo"
      class="c1"
      src="/img/logo.svg"
    />
    <section
      class="c2"
    />
  </main>
</div>
`
